### PR TITLE
Preliminary migration to dcrlibwallet/master multiwallet feature

### PR DESCRIPTION
**Pull request by itswisdomagain**
The aim of this PR is primarily to get dcrios to build with dcrlibwallet/master. This will make implementation of individual features smoother following the v2 mockup.

Ready for review but depends on https://github.com/raedahgroup/dcrlibwallet/pull/81. That needs to be reviewed and merged first before this gets merged.

### CODE REVIEW

File: Decred Wallet/Features/App Launch/StartScreenViewController.swift [hunk 1]
```diff
@@ -60,47 +60,56 @@ class StartScreenViewController: UIViewController {
...
-        else if StartupPinOrPassword.pinOrPasswordIsSet() {
-            self.promptForStartupPinOrPassword()
+    }
+    
+    func checkStartupSecurityAndStartApp() {
+        if StartupPinOrPassword.pinOrPasswordIsSet() {
+            self.promptForStartupPinOrPassword(completion: self.openWalletsAndStartApp)
         } else {
-            self.unlockWalletAndStartApp(pinOrPassword: "public") // unlock wallet using default public passphrase
+            self.openWalletsAndStartApp(startupPinOrPassword: "")
```
Comment (macsleven) [inline Decred Wallet/Features/App Launch/StartScreenViewController.swift]: The default passphrase should be "public". the app crash when you try to unlock an existing wallet

File: Decred Wallet/Features/App Launch/StartScreenViewController.swift [hunk 2]
```diff
@@ -60,47 +60,56 @@ class StartScreenViewController: UIViewController {
...
     
-    func displayWalletSetupScreen() {
-        let walletSetupController = WalletSetupViewController.instantiate().wrapInNavigationcontroller()
-        walletSetupController.isNavigationBarHidden = true
-        AppDelegate.shared.setAndDisplayRootViewController(walletSetupController)
+    func checkStartupSecurityAndLinkExistingWallet() {
+        if StartupPinOrPassword.pinOrPasswordIsSet() {
+            self.promptForStartupPinOrPassword(completion: AppDelegate.walletLoader.linkExistingWalletAndStartApp)
+        } else {
+            AppDelegate.walletLoader.linkExistingWalletAndStartApp(startupPinOrPassword: "")
```
Comment (macsleven) [inline Decred Wallet/Features/App Launch/StartScreenViewController.swift]: The default passphrase should be "public". the app crash when you try to unlock an existing wallet

File: Decred Wallet/Features/Wallet Utils/StartupPinOrPassword.swift [hunk 3]
```diff
@@ -9,10 +9,9 @@
 import Foundation
 import UIKit
 import JGProgressHUD
+import Dcrlibwallet
 
 struct StartupPinOrPassword {
-    static let defaultPublicPassphrase = "public"
```
Comment (macsleven) [inline Decred Wallet/Features/Wallet Utils/StartupPinOrPassword.swift]: why is this removed

File: Decred Wallet/Features/Wallet Utils/StartupPinOrPassword.swift [hunk 4]
```diff
@@ -94,27 +93,28 @@ struct StartupPinOrPassword {
             let newSecurityType = securityType!.lowercased()
             progressHud = Utils.showProgressHud(withText: String(format: LocalizedStrings.changingStartupPINPass, newSecurityType))
         }
-        
-        let currentPublicPassphrase = currentPassword ?? self.defaultPublicPassphrase
-        let newPublicPassphrase = newPinOrPassword ?? self.defaultPublicPassphrase
```
Comment (macsleven) [inline Decred Wallet/Features/Wallet Utils/StartupPinOrPassword.swift]: default password should be retained

File: Decred Wallet/Features/Wallet Utils/StartupPinOrPassword.swift [hunk 5]
```diff
@@ -94,27 +93,28 @@ struct StartupPinOrPassword {
...
-        
+
+        let currentPublicPassphrase = currentPassword ?? ""
+        let newPublicPassphrase = newPinOrPassword ?? ""
```
Comment (macsleven) [inline Decred Wallet/Features/Wallet Utils/StartupPinOrPassword.swift]: The default passphrase is not an empty string

File: Decred Wallet/Features/App Launch/StartScreenViewController.swift (hunk 1 above)
Comment (itswisdomagain) [reply Decred Wallet/Features/App Launch/StartScreenViewController.swift]: Yes, the crash is an issue with dcrlibwallet that I've pushed a fix for. The default public passphrase for wallets is "public" whereas the startup passphrase has no default (or the default is ""). Previously, the startup passphrase was the same as the wallet's public passphrase but they've now been separated in dcrlibwallet.

File: Podfile.lock [hunk 6]
```diff
@@ -30,4 +30,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 03f63fc5c2192d95f913bc200243257155795d2c
 
-COCOAPODS: 1.8.4
+COCOAPODS: 1.8.3
```
Comment (macsleven) [inline Podfile.lock]: Please upgrade your Pod

File: Decred Wallet/Features/Wallet Utils/StartupPinOrPassword.swift [hunk 7]
```diff
@@ -82,14 +82,15 @@ struct StartupPinOrPassword {
...
 
         DispatchQueue.global(qos: .userInitiated).async {
             do {
-                let oldPublicPass = (currentPublicPassphrase as NSString).data(using: String.Encoding.utf8.rawValue)!
-                let newPublicPass = (newPublicPassphrase as NSString).data(using: String.Encoding.utf8.rawValue)!
-                try AppDelegate.walletLoader.wallet?.changePublicPassphrase(oldPublicPass, newPass: newPublicPass)
+                let passphraseType = securityType == SecurityViewController.SECURITY_TYPE_PASSWORD ? DcrlibwalletPassphraseTypePass : DcrlibwalletPassphraseTypePin
+                try AppDelegate.walletLoader.multiWallet.changeStartupPassphrase(currentPublicPassphrase.utf8Bits,
```
Comment (macsleven) [inline Decred Wallet/Features/Wallet Utils/StartupPinOrPassword.swift]: This functions returns the error "invalid_passphrase" and when you try to open the app it crashes with the same error
Screenshot 2020-01-16 at 12 47 43
